packages/block-editor/src/components/block-preview/auto.js

```diff
@@ -33,6 +34,18 @@ function AutoBlockPreview( { viewportWidth, __experimentalPadding } ) {
 		};
 	}, [] );
 
+	// Avoid scrollbars for pattern previews.
+	const editorStyles = useMemo( () => {
+		if ( styles ) {
+			return [
+				...styles,
+				{ css: 'body{overflow:hidden;}', __unstableType: 'presets' },
```

Comment:
I left a comment over here, but just wanted to make sure you see it: https://github.com/WordPress/gutenberg/issues/37573#issuecomment-1019923503

I think if we need to override a style in the iframe, it'd be good to also do something like `height: unset`, the reason being that `height: 100%` is what's causing the scrollbar for patterns.

I'm not sure if you also need `overflow:hidden` to fix the other issue. It seems ok in conjunction with overriding the height, but just `overflow:hidden` on its own causes patterns to be cropped unnecessarily.

Reply:
Here's a comparison to show what I mean:

**Just `overflow: hidden` on the preview body:**
![Screen Shot 2022-01-24 at 6 13 00 pm](https://user-images.githubusercontent.com/677833/150764185-3d25f7ad-df01-4598-9338-b0a0eb2d3fef.png)

**`height: unset` and optionally `overflow: hidden` on the preview body:**
![Screen Shot 2022-01-24 at 6 11 40 pm](https://user-images.githubusercontent.com/677833/150764323-e6e8c6e4-e62e-45c4-9e6c-5b7e8cbcb50f.png)

It's subtle, but you can see for the second one there's the correct spacing at the bottom of the preview when unsetting the height. My concern is that some patterns may be more affected than others by this effect.

Reply:
Thanks, @talldan.

Added the rule to unset height.